dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/CurrencyDataPoint.kt

```diff
@@ -1,21 +1,19 @@
 package org.dataland.datalandbackend.model
 
 import com.fasterxml.jackson.annotation.JsonProperty
+import org.dataland.datalandbackend.interfaces.ExtendedDataPointInterface
 import org.dataland.datalandbackend.model.enums.data.QualityOptions
+import java.math.BigDecimal
 
 /**
  * --- API model ---
  * Fields of a generic data point with unit and its source
  */
-data class DataPointWithUnit<T>(
-    val value: T? = null,
-
+data class CurrencyDataPoint(
+    override val value: BigDecimal? = null,
     @field:JsonProperty(required = true)
     override val quality: QualityOptions,
-
-    override val dataSource: CompanyReportReference? = null,
-
     override val comment: String? = null,
-
-    val unit: String?,
-) : DataPointBaseInformation(quality, dataSource, comment)
+    override val dataSource: ExtendedDocumentReference? = null,
+    val currency: String? = null,
```

Comment:
Does it make sense to allow **currency** data points without specifing a currency?

Reply:
Not changed as discussed in a call